### ISSUE TITLE
Non target faces altered anyway

### ISSUE THREAD

**Issue by a3nima** [glucauze/sd-webui-faceswaplab#7]
Faces that are NOT the target image are still being altered. Checked with img2img and inpaint on latest branch A1111/faceswaplab.

The target face is being changed and this feature works as intended. But there are people in the background and around the scene, there images are not changed like the target image but some small changes happen to them. Its especially obvious in my case because theyre in a depth of field area, not sharp and after faceswap there are small parts around or on them that are being sharpened, though it's NOT a non-intended faceswap. 

I tried to circumvent the problem by going to inpainting and masking the face that should be swapped only, but as I expected, this is overriden and the result is the same.

**Comment by glucauze** [comment]
Okay, either I didn't understand exactly what the problem was. Either the program doesn't behave as you'd expect (in which case it's a documentation or enhancement issue).

Here's an example with inpainting :

![tmpin73r6q7](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/941c9db0-c7fb-487b-b6b1-7aaf8bfbe0d0)
![00012-3581328725-swapped](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/d6882c76-bc97-4158-933d-7b01e0228a0f)
![00011-3581328725](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/9abc91d6-e651-4e93-b1f6-ee1a97e85118)

**Comment by a3nima** [comment]
hi. thanks for your answer. adetailer was the last process in the making of the image. but maybe its codeformer that has done that. the target image worked well but other faces have some sharpened parts on them. I'll try if it occurs without codeformer too. 
I set inpainting with masked only still it happened. what inpainting settings have you used, and did you use the standard img2img inpainting (tab) ?

**Comment by glucauze** [comment]
Here are the details of the previous generation (I think). I did not used any prompt.

Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 3581328725, Size: 512x512, Model hash: bc7da767df, Model: consistent_inpainting.inpainting, Denoising strength: 0.75, Conditional mask weight: 1.0, Mask blur: 4, CFG Rescale phi: 0, Version: v1.5.1, similarity = {}, ref_similarity = {}

I did not use any postprocessing : 

![Screenshot 2023-07-30 at 23-23-59 Stable Diffusion](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/4ef59480-0b60-46b3-93cf-95d33c8caf20)

The idea of post-processing was to improve the poor results of insightface. I hardly use it anymore (except for the post-inpainting part, which also affects the whole image).

I prefer to use the upscaled inswapper which only affects the targetted face.

![Screenshot 2023-07-30 at 23-26-05 Stable Diffusion](https://github.com/glucauze/sd-webui-faceswaplab/assets/137925069/32e37a39-d52a-42df-9edf-3b02348953f2)

You really have to play with the parameters, because what works for one image doesn't always work for another.

**Comment by a3nima** [comment]
yes, it's the codeformer causing the problem. if it could be prevented, would be nice, maybe in a future version. and thanks again!

**Comment by a3nima** [comment]
> I prefer to use the upscaled inswapper which only affects the targetted face.
> 
> ![Screenshot 2023-07-30 at 23-26-05 Stable Diffusion](https://user-images.githubusercontent.com/137925069/257076935-32e37a39-d52a-42df-9edf-3b02348953f2.png)
> 
> You really have to play with the parameters, because what works for one image doesn't always work for anothe

I dont have any of these options (swapper /inswapper) in my faceswaplab extension. Am I blind ??? :D

**Comment by a3nima** [comment]
on the first try with a close up photo the result without inswapper and only codeformer in postprocessing (i think thats what roop does too), was more convincing. but I'll test further

**Comment by glucauze** [comment]
The upscaled inswapper is available in sd settings (the global settings > FaceSwapLab).

It should be better, but it depends on what you do :)

**Comment by a3nima** [comment]
I did not find it to work as well in my test (1. codeformer, 2. inswapper with your settings). 
![grafik](https://github.com/glucauze/sd-webui-faceswaplab/assets/44588648/a79ed60d-8fed-4f46-8c9f-8c0f8b72b338)

**Comment by glucauze** [comment]
If you are using codeformer in inswapper, you should disable codeformer in postprocessing. Otherwise you will have 2 pass of codeformer which might be a bit too much. I think it is what happened here.

Yes SwinIR 4x is worse than LDSR or the one i am using i think. You should try different ones to see what is the best for you
